drizzlepac/haputils/mvm_quality_analysis.py

```diff
@@ -247,6 +266,740 @@ def run_quality_analysis(total_obj_list,
+        and a value of '1' indicates that the second svm_sourcelist_list sourcelist was used as the
+        comparison.
+
+    matched_lines_ref : list
+        indices of the reference catalog that crossmatch to the corresponding element in matched_lines_comp.
+
+    matched_lines_ref : list
+        indices of the comparison catalog that crossmatch to the corresponding element in matched_lines_ref.
+
+    cross_match_details : list
+        three-element list containing the number suitable ref and comp sources available for crossmatch, and
+        the number of sources actually crossmatched
+    """
+    log.setLevel(log_level)
+    # The sourcelist with more sources is automatically set as the "reference" catalog that the other
+    # sourcelist (aka the "comparison" catalog) will be crossmatched to
+    if len(svm_sourcelist_list[0]) >= len(svm_sourcelist_list[1]):
+        ref_index = 0
+        comp_index = 1
+    else:
+        ref_index = 1
+        comp_index = 0
+
+    log.info("Reference catalog: {}, {} sources".format(os.path.basename(overlap_info["svm_sourcelist_{}".format(str(ref_index))]), len(svm_sourcelist_list[ref_index])))
+    log.info("Comparison catalog: {}, {} sources".format(os.path.basename(overlap_info["svm_sourcelist_{}".format(str(comp_index))]), len(svm_sourcelist_list[comp_index])))
+
+    # prepare xyxymatch inputs
+    ref_xy = np.stack((svm_sourcelist_list[ref_index]['X-Skycell'], svm_sourcelist_list[ref_index]['Y-Skycell']), axis=1)
+    comp_xy = np.stack((svm_sourcelist_list[comp_index]['X-Skycell'], svm_sourcelist_list[comp_index]['Y-Skycell']), axis=1)
+    
+    # execute crossmatch
+    matches = xyxymatch(comp_xy, ref_xy, tolerance=5.0, separation=1.0)
+
+    # Report number and percentage of the total number of detected ref and comp sources that were matched
+    log.info("Sourcelist Matching Results")
+    log.info("Reference sourcelist:  {} of {} total sources matched ({} %)".format(len(matches), len(ref_xy), 100.0 * (float(len(matches)) / float(len(ref_xy)))))
+    log.info("Comparison sourcelist: {} of {} total sources matched ({} %)".format(len(matches), len(comp_xy), 100.0 * (float(len(matches)) / float(len(comp_xy)))))
+
+    # extract indices of the matching ref and comp lines and then use them to compile lists of matched X, Y,
+    # RA and DEC for calculation of differences
+    matched_lines_comp = []
+    matched_lines_ref = []
+    for item in matches:
+        matched_lines_comp.append(item[2])
+        matched_lines_ref.append(item[5])
+    cross_match_details = [len(ref_xy), len(comp_xy), len(matches)]
+    return ref_index, comp_index, matched_lines_ref, matched_lines_comp, cross_match_details
+
+# ------------------------------------------------------------------------------------------------------------
+
+
+def determine_if_overlaps_exist(total_obj_list, log_level=logutil.logging.NOTSET):
+    """determines if there are any regions where observations overlap.
```

Comment:
Very nice function that could really be useful in multiple places.

Reply:
thanks!